sim/hunter/items.go

```diff
@@ -569,6 +570,124 @@ func init() {
+			ProcMask:   core.ProcMaskRanged,
+			ProcChance: 0.1,
+			ExtraCondition: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) bool {
+				return !hasMeleeSpecialist
+			},
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if noGainAura.IsActive() {
+					return
+				}
+				stacksAura.Activate(sim)
+				stacksAura.AddStack(sim)
+			},
+		})
+
+		core.MakeProcTriggerAura(&hunter.Unit, core.ProcTrigger{
+			Name:       "Strand Of Fate - Melee Trigger",
+			Callback:   core.CallbackOnSpellHitDealt,
+			Outcome:    core.OutcomeLanded,
+			ProcMask:   core.ProcMaskMelee,
+			ProcChance: 0.1,
+			ExtraCondition: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) bool {
+				return hasMeleeSpecialist
+			},
+			Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+				if noGainAura.IsActive() {
+					return
+				}
+				stacksAura.Activate(sim)
+				stacksAura.AddStack(sim)
+			},
+		})
+
+		// When using the melee version and activating the bow your next strike applies a special Serpent Sting effect.
+		var strands int32
+		SerpentStrikeAura := hunter.GetOrRegisterAura(core.Aura{
+			Label:    "Strand Of Fate - Serpent Sting on Next Strike",
+			ActionID: core.ActionID{SpellID: 1232976},
+			Duration: time.Second * 20,
+			OnGain: func(aura *core.Aura, sim *core.Simulation) {
+				strands = stacksAura.GetStacks()
+			},
+			OnApplyEffects: func(aura *core.Aura, sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				if spell.Matches(ClassSpellMask_HunterStrikes) {
+					hunter.SoFSerpentSting[strands].Cast(sim, target)
+					aura.Deactivate(sim)
+				}
+			},
+		})
+
+		// Gain 40 strength per stack of Strand of Fate consumed
+		strengthAura := hunter.NewTemporaryStatsAura("Strand Of Fate - Strength", core.ActionID{SpellID: 1232969}, stats.Stats{stats.Strength: float64(40 * stacksAura.GetStacks())}, time.Second * 20)
+
+		// The bow active
+		spell := hunter.GetOrRegisterSpell(core.SpellConfig{
+			ActionID: core.ActionID{SpellID: 1231604},
+			Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagOffensiveEquipment,
+			Cast: core.CastConfig{
+				CD: core.Cooldown{
+					Timer:    hunter.NewTimer(),
+					Duration: time.Second * 80,
+				},
+			},
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				if hasMeleeSpecialist && stacksAura.IsActive() {
+					strengthAura.Activate(sim)
+					SerpentStrikeAura.Activate(sim)
+					stacksAura.Deactivate(sim)
+					noGainAura.Activate(sim)
+				} else {
+					if !stacksAura.IsActive() {
+						stacksAura.Activate(sim)
+					}
+					stacksAura.AddStacks(sim, 3)
+					noGainAura.Activate(sim)
+				}
+			},
+		})
+
+		hunter.AddMajorCooldown(core.MajorCooldown{
+			Spell: spell,
+			Type:  core.CooldownTypeDPS,
+		})
+
+		hunter.ItemSwap.RegisterActive(StringsOfFate)
+		hunter.ItemSwap.RegisterProc(StringsOfFate, stacksAura)
```

Comment:
I wonder if the stacks are retained or not when swapping off the bow 🤔 one thing that we'd want to add for sure though is `RegisterProc`s for the two trigger auras up above. Since you put them in conditions you could just add those `RegisterProc`s within their respective blocks